src/patternfly/components/Page/page.scss

```diff
@@ -475,6 +475,10 @@ $pf-page-v6--height-breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg
   flex-shrink: 0;
 }
 
-.#{$page}__drawer > .#{$drawer} {
-    flex: 1 0 auto;
+.#{$page}__drawer .#{$page}__main-container {
+  align-self: revert;
```

Comment:
default for `.#{$page}__main-container` is `align-self: start` which pushes that element to the top of the page component's grid layout. Since `.#{$page}__main-container` goes in `.#{$drawer}__content` (a flex column), that pushes `__main-container` to the left. This can just be the default/stretch.